File: LeetCode/Medium/34. Find First and Last Position of Element in Sorted Array.py
class Solution:
    def searchRange( self, nums: List[int], target: int ) -> List[int]:
        n = len( nums )
        
        if n == 0:
            return [ -1, -1 ]
        
        i = j = bisect_left( nums, target )
        
        if i == n or nums[i] != target:
            return [ -1, -1 ]
        
        while j < len( nums ) and nums[j] == target:
            j += 1
            
        return  [ i, j - 1 ]

File: LeetCode/Medium/34. Find First and Last Position of Element in Sorted Array.cs
public class Solution
{
    
    int[] ans = [ -1, -1 ];
    int[] nums;
    int i, j, k, n, t;
    
    
    
    void f()
    {
        if ( i >= j )
        {
            if ( i < n && nums[i] == t )
            {
                 ans[0] = i;
            }
            
            return;
        }
        
        k = i + ( j - i ) / 2;
        
        if ( nums[k] < t )
        {
            i = k + 1;
            f();
        }
        else
        {
            j = k;
            f();
        }
    }
    
    
    
    void g()
    {
        if ( i == n )
        {
            ans[1] = i - 1;
            
            return;
        }
        
        k = i + ( n - i ) / 2;
        
        if ( nums[k] > t )
        {
            n = k;
            g();
        }
        else
        {
            i = k + 1;
            g();
        }
    }
    
    
    
    public int[] SearchRange( int[] nums, int target )
    {
        this.nums = nums;
        t = target;
        n = nums.Length;
        j = n - 1;
        f();
        
        if ( ans[0] >= 0 )
        {
            g();
        }
        
        return ans;
    }
    
}
